### PR TITLE
Implement memory_info functions

### PR DESCRIPTION
**Pull request by noahshaw11**
## Overview

Implement `couch_debug:memory_info/1` to facilitate memory leak detections.

## Checklist

- [x] Code is written and works correctly
- [ ] Changes are covered by tests
- [ ] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [ ] A PR for documentation changes has been made in https://github.com/apache/couchdb-documentation

### CODE REVIEW

**Comment by noahshaw11** [comment]
Example:
```
(node1@127.0.0.1)3> couch_debug:memory_info([<0.0.0>,<0.1.0>,<0.2.0>,<0.3.0>,<0.4.0>,<0.5.0>,<0.6.0>]).
[{<0.0.0>,"init[<0.0.0>]",
  #{binary => 0,dictionary => 0,heap_size => 2586,links => 3,
    memory => 29568,message_queue_len => 0,monitored_by => 0,
    monitors => 0,stack_size => 3,total_heap_size => 3577}},
 {<0.1.0>,"erts_code_purger[<0.1.0>]",
  #{binary => 0,dictionary => 0,heap_size => 233,links => 0,
    memory => 3328,message_queue_len => 0,monitored_by => 0,
    monitors => 0,stack_size => 2,total_heap_size => 312}},
 {<0.2.0>,"erts_literal_area_collector:start/0[<0.2.0>]",
  #{binary => 0,dictionary => 0,heap_size => 233,links => 0,
    memory => 2696,message_queue_len => 0,monitored_by => 0,
    monitors => 0,stack_size => 6,total_heap_size => 233}},
 {<0.3.0>,
  "erts_dirty_process_signal_handler:start/0[<0.3.0>]",
  #{binary => 0,dictionary => 0,heap_size => 233,links => 0,
    memory => 2696,message_queue_len => 0,monitored_by => 0,
    monitors => 0,stack_size => 3,total_heap_size => 233}},
 {<0.4.0>,
  "erts_dirty_process_signal_handler:start/0[<0.4.0>]",
  #{binary => 0,dictionary => 0,heap_size => 233,links => 0,
    memory => 2696,message_queue_len => 0,monitored_by => 0,
    monitors => 0,stack_size => 3,total_heap_size => 233}},
 {<0.5.0>,
  "erts_dirty_process_signal_handler:start/0[<0.5.0>]",
  #{binary => 0,dictionary => 0,heap_size => 233,links => 0,
    memory => 2696,message_queue_len => 0,monitored_by => 0,
    monitors => 0,stack_size => 3,total_heap_size => 233}},
 {<0.6.0>,"prim_file:start/0[<0.6.0>]",
  #{binary => 0,dictionary => 0,heap_size => 233,links => 0,
    memory => 2880,message_queue_len => 0,monitored_by => 0,
    monitors => 0,stack_size => 2,total_heap_size => 245}}]
```

File: src/couch/src/couch_debug.erl [hunk 1]
```diff
@@ -273,6 +276,20 @@ couch_index_server[<0.288.0>]                |   478240   |         0         |
...
+help(memory_info) ->
+    io:format("
+        - memory_info(ProcessList)
+        - memory_info(ProcessList, InfoKeys)
+        - memory_info(Pid, InfoKeys)
+        --------------------------------
+
+        Obtains the values for a set of optional InfoKeys for each process in ProcessList.
+          - ProcessList: List of processes
+          - InfoKeys: List of desired keys to obtain values for
```
Comment (iilyak) [inline src/couch/src/couch_debug.erl:290]: Please specify which keys are supported.

File: src/couch/src/couch_debug.erl [hunk 2]
```diff
@@ -273,6 +276,20 @@ couch_index_server[<0.288.0>]                |   478240   |         0         |
...
+          - Pid: Initial Pid to start from
+
```
Comment (iilyak) [inline src/couch/src/couch_debug.erl:292]: Please describe output format

**Review by iilyak — APPROVED** [review]
+1

**Pull request merged**